prost-derive/src/lib.rs

```diff
@@ -88,19 +87,18 @@ fn try_message(input: TokenStream) -> Result<TokenStream, Error> {
     // TODO: This encodes oneof fields in the position of their lowest tag,
     // regardless of the currently occupied variant, is that consequential?
     // See: https://developers.google.com/protocol-buffers/docs/encoding#order
-    fields.sort_by_key(|&(_, ref field)| field.tags().into_iter().min().unwrap());
+    fields.sort_by_key(|(_, field)| field.tags().into_iter().min().unwrap());
     let fields = fields;
 
-    let mut tags = fields
+    if let Some((duplicate_tag, _)) = fields
         .iter()
-        .flat_map(|&(_, ref field)| field.tags())
-        .collect::<Vec<_>>();
-    let num_tags = tags.len();
-    tags.sort_unstable();
-    tags.dedup();
-    if tags.len() != num_tags {
-        bail!("message {} has fields with duplicate tags", ident);
-    }
+        .flat_map(|(_, field)| field.tags())
+        .sorted_unstable()
+        .tuple_windows()
+        .find(|(a, b)| a == b)
```

Comment:
it would kinda be preferable to write this as `.duplicates().next()` using `Itertools`, but that method unfortunately requires `use_std`.

Reply:
As I understand prost_derive will run on the host, so it is not subject to `no_std` rules. The code does also `use std::fmt;`, so `std` must be available. So I think it is safe to `use_std` for Itertools. Do you agree?

Reply:
yeah from what i've seen since i published this that is totally possible. i'll update it